kratos/processes/find_intersected_geometrical_objects_process.h

```diff
@@ -7,355 +7,667 @@
+     */
+    virtual void SetOctreeBoundingBox();
+
+    /**
+     * @brief This method marks if intersected
+     * @param rEntity1 The entity of interest
+     * @param rLeaves The Octree cells vectors
+     */
+    virtual void MarkIfIntersected(
+        TEntity& rEntity1,
+        OtreeCellVectorType& rLeaves
+        );
+
+    /**
+     * @brief This method check if there is an intersection between two geometries
+     * @param rFirstGeometry The first geometry
+     * @param rSecondGeometry The second geometry
+     */
+    virtual bool HasIntersection(
+        GeometryType& rFirstGeometry,
+        GeometryType& rSecondGeometry
+        );
+
+    /**
+     * @brief This method check if there is an intersection between two geometries in 2D
+     * @param rFirstGeometry The first geometry
+     * @param rSecondGeometry The second geometry
+     */
+    virtual bool HasIntersection2D(
+        GeometryType& rFirstGeometry,
+        GeometryType& rSecondGeometry
+        );
+
+    /**
+     * @brief This method check if there is an intersection between two geometries in 3D
+     * @param rFirstGeometry The first geometry
+     * @param rSecondGeometry The second geometry
+     */
+    virtual bool HasIntersection3D(
+        GeometryType& rFirstGeometry,
+        GeometryType& rSecondGeometry
+        );
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief Returns the current working space dimension
+     * @return The current working space dimension
+     */
+    std::size_t WorkingSpaceDimension();
+
+    /**
+     * @brief This method generates a new Octree class
+     */
+    void GenerateOctree();
+
+    /**
+     * @brief This method generates a new Octree class (elements)
+     */
+    void GenerateOctreeElements();
+
+    /**
+     * @brief This method generates a new Octree class (conditions)
+     */
+    void GenerateOctreeConditions();
+
+    /**
+     * @brief This method finds intected skin objects
+     * @param rEntity1 The entity of interest
+     * @param rLeaves The Octree cells vectors
+     * @param rResults The expected results
+     */
+    void FindIntersectedSkinObjects(
+        TEntity& rEntity1,
+        OtreeCellVectorType& rLeaves,
+        PointerVector<GeometricalObject>& rResults
+        );
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    Parameters GetDefaultParameters();
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    FindIntersectedGeometricalObjectsProcess& operator=(FindIntersectedGeometricalObjectsProcess const& rOther);
```

Comment:
```suggestion
    FindIntersectedGeometricalObjectsProcess& operator=(FindIntersectedGeometricalObjectsProcess const& rOther) = delete;
```
And move it to the life cycle section

Reply:
Any reason to avoid deleting it?

Reply:
I added manually, don't trust GitHub